app/models/listing.py

```diff
@@ -65,34 +65,50 @@ def category_name(self):
     @staticmethod
     def search(**kwargs):
         """ Returns all listings matching the criteria """
-        filter_list = []
+        name = "%"
+        vendor = "%"
         if 'main_search_term' in kwargs:
             term = kwargs['main_search_term']
-            filter_list.append(or_(
-                Listing.name.like('%{}%'.format(term)),
-                Listing.description.like('%{}%'.format(term)))
-            )
+            name = '%{}%'.format(term)
+
         if 'name_search_term' in kwargs:
             term = kwargs['name_search_term']
-            filter_list.append(or_(
-                Vendor.first_name.like('%{}%'.format(term)),
-                Vendor.last_name.like('%{}%'.format(term)),
-                Vendor.company_name.like('%{}%'.format(term)))
-            )
-        if 'available' in kwargs:
-            filter_list.append(Listing.available == kwargs['available'])
+            vendor = '%{}%'.format(term)
 
+        price_min = 0
+        # price_max some arbitrarily large number to ensure all price are shown unless a user specifies a price
+        price_max = 1000000
```

Comment:
A couple thought: Before with `filter_list` we wouldn't need to do this. As a rule of thumb, the more complex a query is, the slower it'll be, so the more we can leave out the better. And if were going to choose some arbitrarily large number, you're better off with `float('inf')`, which is Python's `infinity`.